applications/nftaa.md

```diff
@@ -0,0 +1,411 @@
+|     1e. | Get owned assets                             | Functionality needed to get all assets owned by given NFTAA |
+|     1f. | Get NFTAAs in account and owner              | Functions for getting all assets owned by given NFTAA and to get owner of given NFTAA with metadata |
+
+### Milestone 2 — Additional features
+
+- **Estimated Duration:** 2 month
+- **FTE:**  1.5
+- **Costs:** 9,000 USD
+
+Functionalities from pallet_nfts:
+- pallet::dispatchables::approve_item_attributes
+- pallet::dispatchables::approve_transfer
+- pallet::dispatchables::cancel_approval
+- pallet::dispatchables::cancel_item_attributes_approval
+- pallet::dispatchables::cancel_swap
+- pallet::dispatchables::claim_swap
+- pallet::dispatchables::clear_all_transfer_approvals
+- pallet::dispatchables::create_swap
+- pallet::dispatchables::force_collection_config
+- pallet::dispatchables::force_collection_owner
+- pallet::dispatchables::force_create
+- pallet::dispatchables::force_mint
+- pallet::dispatchables::force_set_attribute
+- pallet::dispatchables::mint_pre_signed
+- pallet::dispatchables::pay_tips
+- pallet::dispatchables::set_accept_ownership
+
+After these, we will add our own functions related to NFTAA staking functionality.
+
+| Number | Deliverable             | Specification                                                                                                                      |
+|-------:|-------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| **0a.** | License                                      | MIT |
+| **0b.** | Documentation                                | Inline documentation of code, as well as startup configuration with all necessary commands, included in repository |
+| **0c.** | Testing and Testing Guide                    | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker                                       | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+|    2a. | nftaa_check             | We will add functionalities to read and check if NFTAA exists for a given account                                                  |
+|    2b. | nftaa_collections       | We will add functionalities to read existing collections and manage NFTAAs if they are in a collection.                            |
+|    2c. | nftaa_stake/unstake     | The functionality needed to do stake/unstake operations on behalf of NFTAA, its like deposit/withdrawal functions                  |
+|    2d. | nftaa_increase/decrease | Functions needed to increase or decrease stake by NFTAA, this is needed for full compatibility with traditional methods of staking |
```

Comment:
I'm confused by this milestone. Previously you described your project as a solution to use the NFTAAs as virtual keyless accounts which are able to do anything a regular account can. Yet this milestone sounds like you are putting in a lot of work to make your solution compatible with the existing pallets for staking. If this is indeed necessary I think you should try to find a more generic solution which doesn't create this overhead for every pallet you want to support with NFTAAs.

Reply:
We don't want to do this for every pallet and we certainly want our solution to be generic, but the case of staking is special and therefore we want to have these functionalities explicitly ready. But it still applies NFTAAs are virtual keyless accounts which are able to do anything a regular account can.

Reply:
What makes staking special?

Reply:
We believe staking is an ideal use case to showcase how NFTAA operates because it's a concept that many users are already familiar with, making it easier to grasp the potential of NFTAA. Additionally, staking plays a crucial role in maintaining network security, making it a fitting example to demonstrate the robustness and functionality of NFTAA in a practical environment.

Reply:
I know what staking ist. I just don't understand your answer. If you are planning to make your solution generic as in "this NFTAA can be used just like a regular account", then why do you need to put in any work to make your solution compatible with the staking pallet. I hope you understand where my issue is here. Staking is already supported for regular accounts. With you adding just another layer on top of accounts with this form of account abstraction I don't understand why you would need to modify anything below this account level. E.g. How individual pallets work with accounts.

Reply:
I see. Then just quickly could you please describe how this showcase will look like? Will it be a webapp that you create so that users can conveniently interact with their NFTAAs for staking? I initially understood this milestone in a way that you are planning to introduce additional staking related logic within your NFTAA pallet. I wouldn't like this for the aforementioned reasons in my earlier comments. If you are planning to showcase NFTAAs for staking with a web-app unrelated to the pallet, then I'd be fine with that.

Reply:
I rearranged it all in my head a little bit and I think we finally found the same point of view. It will be as you write we will use staking as a showcase using the web app but we won't add any specific functionality to the nftaa_pallet itself for this functionality. So our generic proxy functionality will be used for staking. For that reason I have removed this parts from the milestone:
- 2c. nftaa stake/unstake
- 2d. nftaa increase/decrease

Thank you for your patience in this subdiscussion :)

Reply:
got it, nice!